packages/@aws-cdk/aws-autoscaling/lib/lifecycle-hook.ts

```diff
@@ -73,18 +75,36 @@ export interface LifecycleHookProps extends BasicLifecycleHookProps {
 export interface ILifecycleHook extends IResource {
   /**
    * The role for the lifecycle hook to execute
+   *
+   * @default - A default role is created if 'notificationTarget' is specified.
+   * Otherwise, no role is created.
    */
-  readonly role: iam.IRole;
+  role: iam.IRole;
```

Comment:
I don't think this should become read/write.

I know the API doesn't let you do what you need to do, which sucks. We'll have to find a way around that, but this is not the right way. We're going to either (a) have to break the api or (b) do type hackery.

Let's do (a), this once.

Reply:
Do you mean removing this logic
```
private readonly _role?: iam.IRole;

public get role(): iam.IRole {
  if (!this._role) { throw new Error('Oh no we don\'t have a role'); }
  return this._role;
}
```
and going back to having `public readonly role?: iam.IRole;` like we had [previously](https://github.com/aws/aws-cdk/pull/16187#discussion_r696420739)?